test/remote/gateways/remote_rapyd_test.rb

```diff
@@ -0,0 +1,176 @@
+require 'test_helper'
+
+class RemoteRapydTest < Test::Unit::TestCase
+  def setup
+    @gateway = RapydGateway.new(fixtures(:rapyd))
+
+    @amount = 100
+    @credit_card = credit_card('4111111111111111', first_name: 'Ryan', last_name: 'Reynolds', month: '12', year: '2035', verification_value: '345')
+    @declined_card = credit_card('4111111111111105')
+    @check = check
+    @options = {
+      type: 'us_visa_card',
+      currency: 'USD'
+    }
+    @ach_options = {
+      type: 'us_ach_bank',
+      currency: 'USD',
+      proof_of_authorization: false,
+      payment_purpose: 'Testing Purpose'
+    }
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'SUCCESS', response.message
+  end
+
+  def test_successful_purchase_with_address
+    options = {
+      address: {
+        name: 'Henry Winkler',
+        address1: '123 Happy Days Lane'
+      }
+    }
+
+    response = @gateway.purchase(@amount, @credit_card, @options.merge(options))
+    assert_success response
+    assert_equal 'SUCCESS', response.message
+  end
+
+  def test_successful_purchase_using_ach
+    response = @gateway.purchase(100000, @check, @ach_options)
+    assert_success response
+    assert_equal 'SUCCESS', response.message
+    assert_equal 'CLO', response.params['data']['status']
```

Comment:
Nitpicky question: Is there any benefit to asserting the value of this `status` response field?

Reply:
Great question and yes! ACH transactions have to go through a multi-step (aka multi-api-call) process to be completed. The `CLO` here means it is CLOSED at the gateway level. If the multi part process had not been successful, it would have a status of `ACT` for ACTIVE. We want to make sure it closes out for purchase, so that's why that status value is being checked.